Controller/Subscriber/Subscribe.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Smaily\SmailyForMagento\Controller\Subscriber;
+
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\RequestInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Smaily\SmailyForMagento\Helper\Data as Helper;
+
+/**
+ * Sends new subscribers from newsletter sign-up form directly to Smaily.
+ */
+class Subscribe
+{
+    /**
+     * Smaily helper class contains all methods.
+     *
+     * @var Smaily\SmailyForMagento\Helper\Data
+     */
+    protected $helper;
+
+    /**
+     * Gathers extra data from store.
+     *
+     * @var Magento\Store\Model\StoreManagerInterface
+     */
+    private $_storeManager;
+
+    /**
+     * Gathers extra data if customer is logged in when subscribing.
+     *
+     * @var Magento\Customer\Model\Session
+     */
+    private $_customerSession;
+
+    /**
+     * Gather post parameters (Name field from form).
+     *
+     * @var Magento\Framework\App\RequestInterface
+     */
+    private $_request;
+
+    public function __construct(
+        Helper $helper,
+        RequestInterface $request,
+        StoreManagerInterface $storeManager,
+        Session $customerSession
+    ) {
+        $this->helper = $helper;
+        $this->_request = $request;
+        $this->_storeManager = $storeManager;
+        $this->_customerSession = $customerSession;
+    }
+
+    /**
+     * Send subscriber data to Smaily.
+     *
+     * @param \Magento\Newsletter\Model\Subscriber $subject
+     * @param string $email
+     * @return null
+     */
+    public function beforeSubscribe(\Magento\Newsletter\Model\Subscriber $subject, $email)
+    {
+        // Check Smaily extension/newsletter subscribers collection are enabled.
+        if ($this->helper->isEnabled() && $this->helper->isNewsletterSubscriptionEnabled()) {
+            $autoresponderId = $this->helper->getGeneralConfig('autoresponder_id');
+            $name = '';
```

Comment:
This functionality can be removed because standard form doesn't have name field value in subscription form. I would keep it though, so that when store owner creates name field in their form our plugin manages to catch that value.

Reply:
Let's remove the functionality and investigate an option where we could capture any data posted into that form.

Reply:
Now getting all form data and sending forwarding to Smaily 79415a3ceb6bb8b6d81d814dfa5c25c7c88ad6d9